pkg/hubble/parser/seven/http.go

```diff
@@ -80,3 +84,36 @@ func (p *Parser) httpSummary(flowType accesslog.FlowType, http *accesslog.LogRec
 	}
 	return ""
 }
+
+// filterHeader receives a key-value pair of an http header along with an HttpHeadersList.
+// Based on the allow/deny lists of the provided HttpHeadersList it returns the original value
+// or the redacted constant "HUBBLE_REDACTED" accordingly:
+//  1. If HttpHeadersList is not nil (meaning that hubble.redact feature is enabled) but both allow/deny lists are empty then the value of the
+//     header is redacted by default.
+//  2. If the header's key is contained in the allow list then the value
+//     of the header will not be redacted.
+//  3. If the header's key is contained in the deny list then the value
+//     of the header will be redacted.
+//  4. If none of the above happens, then if there is any allow list defined then the value will be redacted
+//     otherwise if there is a deny list defined the value will not be redacted.
+func filterHeader(key string, value string, headersList *options.HttpHeadersList) string {
+	if headersList != nil {
+		if len(headersList.Allow) == 0 && len(headersList.Deny) == 0 {
+			// That's the default case, where redact is generally enabled but not headers' lists
+			// have been specified. In that case we redact everything by default.
+			return defaults.SensitiveValueRedacted
+		}
+		if slices.Contains(headersList.Allow, strings.ToLower(key)) {
```

Comment:
This should be a map lookup.

Reply:
Done in [842a01a](https://github.com/cilium/cilium/pull/26724/commits/842a01aa97d28ef5dc003bb657d2ef13cabcf1d4)